radicle-httpd/src/lib.rs

```diff
@@ -52,23 +60,36 @@ pub async fn run(options: Options) -> anyhow::Result<()> {
         .merge(git_router)
         .nest("/api", api_router)
         .nest("/raw", raw_router)
+        .layer(middleware::from_fn(pass_tracing_info_middleware))
         .layer(
             TraceLayer::new_for_http()
-                .make_span_with(|request: &Request<Body>| {
-                    tracing::info_span!(
-                        "request",
-                        method = %request.method(),
-                        uri = %request.uri(),
-                        status = tracing::field::Empty,
-                        latency = tracing::field::Empty,
-                    )
+                .make_span_with(move |_request: &Request<Body>| {
+                    tracing::info_span!("request", id = %id_counter.clone().next())
                 })
                 .on_response(
-                    |response: &Response<BoxBody>, latency: Duration, span: &Span| {
-                        span.record("status", &tracing::field::debug(response.status()));
-                        span.record("latency", &tracing::field::debug(latency));
-
-                        tracing::info!("Processed");
+                    |response: &Response<BoxBody>, latency: Duration, _span: &Span| {
+                        if let Some(info) = response.extensions().get::<TracingInfo>() {
+                            tracing::info!(
+                                "{} \"{} {} {:?}\" {} {:?} {}",
+                                info.connect_info.0,
+                                info.method,
+                                info.uri,
+                                info.version,
+                                ColoredStatus(response.status()),
+                                latency,
+                                Paint::dim(
+                                    response
+                                        .body()
+                                        .size_hint()
```

Comment:
`.size_hint()` returns a lower and upper bound, you can get a number by calling `.exact()` which will return one if they are the same, otherwise `None`